src/main/java/uk/ac/cam/cl/dtg/segue/configuration/SegueGuiceConfigurationModule.java

```diff
@@ -1011,18 +1010,18 @@ private static SegueJobService getSegueJobService(final PropertiesLoader propert
                 configuredScheduledJobs.add(eventFeedbackEmail);
             }
 
-            segueJobService = new SegueJobService(configuredScheduledJobs, database);
-
-            // Simply removing the following jobs from the configuredScheduledJobs wouldn't remove them from the qrtz_job_details
-            // page so they need properly removing.
+            // Simply removing jobs from configuredScheduledJobs won't de-register them if they
+            // are currently configured, so the constructor takes a list of jobs to remove too.
+            List<SegueScheduledJob> scheduledJobsToRemove = new ArrayList<>();
             if (mailjetKey == null && mailjetSecret == null) {
```

Comment:
Ah, yes, that was what I wanted when Ben added it, but they couldn't be because the scheduler needed to have been created between the first if and this one. Now that none of it is being done synchronously this should be possible!

Reply:
This technically changes the logic slightly (added if both non-null, removed if both null ==> added if both non-null, removed if either null), but that seems safest, since we can't actually run the job if either config entry is null anyhow!